src/Schema/Directives/ThrottleDirective.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Closure;
+use GraphQL\Type\Definition\ResolveInfo;
+use Illuminate\Cache\RateLimiter;
+use Illuminate\Cache\RateLimiting\Unlimited;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Nuwave\Lighthouse\Exceptions\DirectiveException;
+use Nuwave\Lighthouse\Exceptions\RateLimitException;
+use Nuwave\Lighthouse\Schema\Values\FieldValue;
+use Nuwave\Lighthouse\Support\Contracts\FieldMiddleware;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+use Symfony\Component\HttpFoundation\Response;
+
+class ThrottleDirective extends BaseDirective implements FieldMiddleware
+{
+    /**
+     * The rate limiter instance.
+     *
+     * @var \Illuminate\Cache\RateLimiter
+     */
+    protected $limiter;
+
+    /**
+     * @var \Illuminate\Http\Request
+     */
+    protected $request;
+
+    /**
+     * Create a new request throttler.
+     */
+    public function __construct(RateLimiter $limiter, Request $request)
+    {
+        $this->limiter = $limiter;
+        $this->request = $request;
+    }
+
+    public static function definition(): string
+    {
+        return <<<'GRAPHQL'
+"""
+Sets rate limit to access the field. Does the same as ThrottleRequests Laravel Middleware.
+"""
+directive @throttle (
+    """
+    Named preconfigured rate limiter. Requires Larave 8.x or later.
+    """
+    name: String
+
+    """
+    Maximum number of attempts in a specified time interval.
+    """
+    maxAttempts: Int
+
+    """
+    Time in minutes to reset attempts.
+    """
+    decayMinutes: Float
+
+    """
+    Prefix to distinguish several field groups.
+    """
+    prefix: String
+
+) on FIELD_DEFINITION
+GRAPHQL;
+    }
+
+    public function handleField(FieldValue $fieldValue, Closure $next): FieldValue
+    {
+        $originalResolver = $fieldValue->getResolver();
+
+        $limits = [];
+        $name = $this->directiveArgValue('name', null);
+        if ($name !== null) {
+            if (! method_exists($this->limiter, 'limiter')) {
+                throw new DirectiveException('Named limiter requires Laravel 8.x or later');
+            }
+
+            $limiter = $this->limiter->limiter($name);
+            /** @phpstan-ignore-next-line $limiter may be null although it's not specified in limiter() PHPDoc */
+            if (is_null($limiter)) {
+                throw new DirectiveException("Named limiter $name is not found.");
+            }
+
+            $limiterResponse = call_user_func($limiter, $this->request);
+            if (class_exists(Unlimited::class) && $limiterResponse instanceof Unlimited) {
```

Comment:
Won't this fail if the class does not exist? Maybe we have to use a string instead.

Reply:
I've removed this check - it is not needed in production environment. This class always exists in Laravel 8